src/main/java/org/sagebionetworks/bridge/reporter/worker/BridgeReporterProcessor.java

```diff
@@ -115,4 +95,27 @@ public void callback(String messageBody) throws IOException, PollSqsWorkerBadReq
                     startDateTime + ",endDate=" + endDateTime);
         }
     }
+
+    /**
+     * Helper method to call getUploadsForStudy distinguishing by daily or weekly
+     * @param studyId
+     * @param startDateTime
+     * @param endDateTime
+     * @param scheduleType
+     * @return
+     */
+    private ResourceList<Upload> getUploadsForStudyHelper(String studyId, DateTime startDateTime, DateTime endDateTime, ReportScheduleName scheduleType) {
+        List<Upload> uploadList = new ArrayList<>();
+
+        if (scheduleType == ReportScheduleName.DAILY) {
+            uploadList = bridgeHelper.getUploadsForStudy(studyId, startDateTime, endDateTime).getItems();
+        } else {
+            while (startDateTime.isBefore(endDateTime)) {
+                uploadList.addAll(bridgeHelper.getUploadsForStudy(studyId, startDateTime, startDateTime.withHourOfDay(23).withMinuteOfHour(59).withSecondOfMinute(59).withMillisOfSecond(999)).getItems());
+                startDateTime = startDateTime.plusDays(1);
+            }
+        }
+
+        return new ResourceList<Upload>(uploadList, uploadList.size());
+    }
```

Comment:
For organizational purposes, this would probably be better as part of BridgeHelper instead of in the processor.

Instead of keying off ReportScheduleName, it might be cleaner to simply check if the date range is greated than a day, and break it down if it is.

The whole point of breaking this down into sub-requests is so we down brown out our DDB capacity. If we call Bridge Server as fast as possible, this defeats the whole purpose. You need some sleep time between each iteration of the loop.

Not sure if this handles weird things like leap seconds. It might be safer to say .plusDays(1).minusMilliseconds(1) instead.